File: backend/src/main/java/com/ethicssra/domain/Comment.java
package com.ethicssra.domain;

import jakarta.persistence.*;
import java.time.Instant;

@Entity
@Table(name = "comments")
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "project_id")
    private Long projectId;

    @Column(name = "requirement_id")
    private Long requirementId;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(name = "parent_id")
    private Long parentId;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @PrePersist
    void prePersist() {
        if (createdAt == null) createdAt = Instant.now();
        if (updatedAt == null) updatedAt = Instant.now();
    }

    @PreUpdate
    void preUpdate() {
        updatedAt = Instant.now();
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getProjectId() { return projectId; }
    public void setProjectId(Long projectId) { this.projectId = projectId; }

    public Long getRequirementId() { return requirementId; }
    public void setRequirementId(Long requirementId) { this.requirementId = requirementId; }

    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }

    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }

    public Long getParentId() { return parentId; }
    public void setParentId(Long parentId) { this.parentId = parentId; }

    public Instant getCreatedAt() { return createdAt; }
    public void setCreatedAt(Instant createdAt) { this.createdAt = createdAt; }

    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}

File: frontend/src/api/comment.ts
import http, { unwrap } from './http'
import type { ApiEnvelope } from './http'

export interface CommentDto {
  id: number
  projectId: number
  requirementId: number
  userId: number
  userName: string
  content: string
  parentId: number | null
  createdAt: string
  updatedAt: string
}

export interface AddCommentRequest {
  projectId: number
  requirementId: number
  content: string
  parentId?: number
}

export const commentApi = {
  list: (projectId: number, requirementId?: number) => {
    const params = requirementId 
      ? `?projectId=${projectId}&requirementId=${requirementId}`
      : `?projectId=${projectId}`
    return unwrap<CommentDto[]>(http.get<ApiEnvelope<CommentDto[]>>(`/comments${params}`))
  },
  
  add: (data: AddCommentRequest) => unwrap<CommentDto>(http.post<ApiEnvelope<CommentDto>>('/comments', data)),
  
  delete: (id: number) => unwrap(http.delete(`/comments/${id}`)),
}